test/fw/ptl/utils/pbs_testsuite.py

```diff
@@ -828,6 +829,294 @@ def init_proc_mon(self):
+        as a string, otherwise None
+        """
+        if conf == "PBS_SERVER":
+            return primary_host
+        elif conf == "PBS_START_SCHED":
+            if hosttype == "server":
+                return "1"
+            else:
+                return "0"
+        elif conf == "PBS_START_COMM":
+            if hosttype == "comm":
+                return "1"
+            else:
+                return "0"
+        elif conf == "PBS_START_SERVER":
+            if hosttype == "server":
+                return "1"
+            else:
+                return "0"
+        elif conf == "PBS_START_MOM":
+            if hosttype == "mom":
+                return "1"
+            else:
+                return "0"
+        elif conf == "PBS_CORE_LIMIT":
+            return "unlimited"
+        elif conf == "PBS_SCP":
+            scppath = self.du.which(hostobj.hostname, "scp")
+            if scppath != "scp":
+                return scppath
+
+        return None
+
+    def revert_pbsconf(self):
+        """
+        Revert the following about the pbs.conf file:
+        - start bits
+        - PBS_SERVER
+        - PBS_SCP
+        - PBS_CORE_LIMIT
+        Also start/stop the appropriate daemons
+        """
+        primary_server = self.servers.values()[0]
+
+        none_vals = {
+            "PBS_HOME": None,
+            "PBS_EXEC": None,
+            "PBS_SERVER": None,
+            "PBS_START_SCHED": None,
+            "PBS_START_COMM": None,
+            "PBS_START_SERVER": None,
+            "PBS_START_MOM": None,
+            "PBS_CORE_LIMIT": None,
+            "PBS_SCP": None
+        }
+
+        # Revert server hosts' pbs.conf
+        for server in self.servers.values():
+            default_vals = none_vals
+            cmds_to_exec = []
+            dmns_to_start = 0
+            pbs_conf_filepath = self.du.get_pbs_conf_file(server.hostname)
+            pbs_conf_val = self.du.parse_pbs_config(server.hostname,
+                                                    pbs_conf_filepath)
+            if not pbs_conf_val or len(pbs_conf_val) < 1:
+                raise ValueError("Could not parse pbs.conf on host %s" %
+                                 (server.hostname))
+
+            # to start with, set all keys in default_vals with values from the
+            # existing pbs.conf
+            for conf in default_vals:
+                try:
+                    default_vals[conf] = pbs_conf_val[conf]
+                except KeyError:
+                    # existing pbs.conf doesn't have a default variable set
+                    # Try to determine the default value for it
+                    val = self._get_dflt_pbsconfval(conf,
+                                                    primary_server.hostname,
+                                                    "server", server)
+                    if val is None:
+                        print "Couldn't revert %s in pbs.conf" % (key)
+                        raise
```

Comment:
when it is raised from the setUp, since we don't catch it, I'm guessing that the test execution will stop and we will bail out of pbs_benchpress. I think this is the right thing to do because if we continue on without a pristine pbs.conf then the results of test execution wouldn't be reliable.